### PR TITLE
Rate limiting depending on user authenticated status

### PR DESCRIPTION
**Pull request by cquintana92**
This PR changes how we are obtaining a the key for the rate limiting process.

- If the user is not authenticated, the request source IP is used.
- If the user is authenticated, the user ID is used.

### CODE REVIEW

**Review by nguyenkims — APPROVED** [review]
I wonder if we can have a test for this change?

File: app/extensions.py [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-limiter = Limiter(key_func=get_remote_address)
 
+# We want to rate limit based on:
+# - If the user is not logged in: request source IP
+# - If the user is logged in: user_id
+def __key_func():
+    if current_user.is_authenticated:
+        return current_user.id
+    else:
+        return get_remote_address()
```
Comment (acasajus) [inline app/extensions.py:16]:
```suggestion
    if current_user.is_authenticated:
        return f"userid:{current_user.id}"
    else:
        ip_addr = get_remote_address()
        return f"ip:{ip_addr}"
```

Comment (acasajus) [reply app/extensions.py:16]: Just to scope a bit the field by which we're rate limiting.

**Comment by cquintana92** [comment]
> I wonder if we can have a test for this change?

I managed to add a few tests that prove the behaviour is the one we want.
It's a bit hacky dealing with the rate_limiter in test time, but I think I managed to add a meaningful explanation to the test file.

**Pull request merged**